airflow/dag_processing/bundles/git.py

```diff
@@ -256,7 +257,15 @@ def refresh(self) -> None:
         with self.lock():
             with self.hook.configure_hook_env():
                 self._fetch_bare_repo()
-                self.repo.remotes.origin.pull()
+                self.repo.remotes.origin.fetch(
```

Comment:
nit: It appears you are handling two separate repositories (a bare repo and a working copy); if I understand this right. It may help to add comments explaining why both fetches are necessary.

Reply:
Also, do we need to retry here?